### PR TITLE
Fixing bug where named-exports didn't work on named registers. Resolves #2073.

### LINKED ISSUE #2073: Named exports limitation - doesn't work for modules in registerRegistry
See https://github.com/joeldenning/systemjs-issue-named-exports-registerRegistry for a demonstration of the problem.

The problem is that the the AMD extra [does not call getRegister for named modules](https://github.com/systemjs/systemjs/blob/master/src/extras/amd.js#L102). The named-exports extra [relies on getRegister](https://github.com/systemjs/systemjs/blob/master/src/extras/named-exports.js#L16) to work.

The demo repo is putting two modules into the systemjs registries - the first is in the "module registry" and has the named-exports extra applied to it. The second is the named module in the "registerRegistry" and it does not have the named-exports extra applied to it. The first time that `System.import()` is called, it always retrieves the module from the module registry, but subsequent calls to `System.import()` always retrieve the module from the registerRegistry.

### PR DESCRIPTION
**Pull request by joeldenning**
Resolves #2073. I verified that this adds 49 bytes to the amd.min.js extra.

### CODE REVIEW

File: src/extras/amd.js [hunk 1]
```diff
@@ -128,4 +133,13 @@
     }
   };
   global.define.amd = {};
+
+  function insertToRegisterRegistry(name, define) {
```
Comment (guybedford) [inline src/extras/amd.js]: Maybe just call this `addToRegisterRegistry`.

File: src/extras/amd.js [hunk 2]
```diff
@@ -10,6 +10,8 @@
     throw Error('AMD require not supported.');
   }
 
+  let tempRegister;
```
Comment (guybedford) [inline src/extras/amd.js]: Hate to be tooo pedantic, but `tmpRegister` please.

**Comment by joeldenning** [comment]
Updated with your feedback

**Comment by guybedford** [comment]
Great, feel free to merge and release.

**Comment by joeldenning** [comment]
Released in https://github.com/systemjs/systemjs/releases/tag/6.1.5

**Pull request merged**